pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +1008,58 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// ShouldUpdateZoneEndpointList checks whether zone endpoint list need to be updated or not
+func ShouldUpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) (bool, error) {
+	if zoneName == "" {
+		return false, errors.Errorf("zone name can't be empty")
+	}
+
+	zoneExists, endpoints := findZoneEndpoints(zoneName, zones)
+	if !zoneExists {
+		return false, nil
+	}
+
+	zoneEndpointListEqual, err := listsAreEqual(desiredEndpointList, endpoints)
+	return !zoneEndpointListEqual, err
+}
+
+func findZoneEndpoints(targetZone string, zones []zoneType) (bool, []string) {
+	for _, z := range zones {
+		if z.Name == targetZone {
+			return true, z.Endpoints
+		}
+	}
+	return false, []string{}
+}
+
+func listsAreEqual(a, b []string) (bool, error) {
+	as := make([]string, len(a))
+	bs := make([]string, len(b))
+	copy(as, a)
+	copy(bs, b)
+	sort.Strings(as)
+	sort.Strings(bs)
+
+	if checkDuplicatesInSortedSlice(as) || checkDuplicatesInSortedSlice(bs) {
+		return false, errors.Errorf("list have duplicate entries")
+	}
+	if len(a) != len(b) {
+		return false, nil
+	}
+	for i := 0; i < len(a); i++ {
+		if as[i] != bs[i] {
+			return false, nil
+		}
+	}
+	return true, nil
+}
```

Comment:
TBH, I consider the `listsAreEqual` to be an extremely low-level function. Error handling the case where there are duplicates seems like it would be good to leave outside the scope of the function to me. 

Outside the context of this PR, if we just consider two lists, `[a, a, a]` and `[a, a]`, we don't necessarily care about duplicates. Obviously they are not equal. But `[a, a]` and `[a, a]` are equal.

In addition, if the user receives an error about the lists having duplicate entries, firstly the error doesn't give additional help to know what lists are being compared by showing their entries. More importantly, however, bubbling this error up to the user isn't really helpful to help them understand what is wrong. 

If the user has mis-configured their endpoints by putting duplicate entries in, we have 2 options:
1. don't care
2. return an error informing them they supplied duplicate entries (IMO, best for user)

If the RGW (for whatever reason) returns duplicate entries, it should be safe for us to squash the list to remove duplicates.